js/views/message_view.js

```diff
@@ -248,11 +256,13 @@
             }
             this.$('.meta .hasRetry').remove();
             if (this.model.hasNetworkError()) {
-                this.$('.meta').prepend(new NetworkErrorView().render().el);
+                var networkErrorView = new NetworkErrorView({model: this.model});
+                this.$('.meta').prepend(networkErrorView.render().el);
             }
             this.$('.meta .some-failed').remove();
             if (this.model.someRecipientsFailed()) {
-                this.$('.meta').prepend(new SomeFailedView().render().el);
+                var someFailedView = new SomeFailedView();
+                this.$('.meta').prepend(someFailedView.render().el);
```

Comment:
Two if's means 4 possible states. So, if I'm reading this right, the different cases for a group are:

1. all recipients failed due to network errors:
  > Message not sent. _Resend_
2. only some recipients failed, and one or more were due to network errors:
  > Some recipients failed _Resend_
3. only some recipients failed, but none due to network errors (i.e., key changes):
  > Some recipients failed *(red error marker beside the message bubble)*
4. all recipients failed, but none due to network errors
  > *(No text in the bubble, just a red error marker beside the bubble)*

Seems reasonable... Maybe add a period to "Some recipients failed" for consistency.